service_configuration_lib/spark_config.py

```diff
@@ -432,12 +436,36 @@ def compute_executor_instances_k8s(user_spark_opts: Dict[str, str]) -> int:
     return executor_instances
 
 
+def _cap_executor_resources(user_spark_opts: Dict[str, str]) -> Dict[str, str]:
+    executor_memory = user_spark_opts.get('spark.executor.memory', DEFAULT_EXECUTOR_MEMORY)
+    executor_cores = int(user_spark_opts.get('spark.executor.cores', str(DEFAULT_EXECUTOR_CORES)))
+
+    memory_mb = parse_memory_string(executor_memory)
+    if memory_mb > EXECUTOR_MEMORY_CAP_GB * 1024:
+        executor_memory = f'{EXECUTOR_MEMORY_CAP_GB}g'
+        log.warning(f'Executor memory is {memory_mb / 1024}g, capped to {executor_memory}')
+    elif memory_mb > EXECUTOR_MEMORY_WARN_GB * 1024:
```

Comment:
Nit: Change the message to something like, "We recommend using setting memory as 28g and executor cores as 4"